packages/headless/src/features/commerce/common/actions.ts

```diff
@@ -62,20 +69,21 @@ export const buildBaseCommerceAPIRequest = async (
       view,
       cart: state.cart.cartItems.map((id) => state.cart.cart[id]),
     },
-    ...(state.commercePagination && {
-      page: state.commercePagination.page,
-      ...(state.commercePagination.perPage && {
-        perPage: state.commercePagination.perPage,
+    ...(state.commercePagination?.[solutionTypeId] && {
+      page: state.commercePagination[solutionTypeId].page,
+      ...(state.commercePagination[solutionTypeId].perPage && {
+        perPage: state.commercePagination[solutionTypeId].perPage,
```

Comment:
If we want to extends the principle to others things, such as facet and the like, should we instead have `state[solutionTypeId].commercePagination.*` ?

Reply:
Because we could have several productListing and recs, right?

Reply:
Yup!

Reply:
Well, we have to support multiple recs slots on the same page.

For listing and search, there should technically only be one per page. However, by supporting multiple recs slots, I think the effort to support multiple listings or multiple search pages would be minimal.

Reply:
I had a go at using multiple reducers for multiple slices: https://github.com/coveo/ui-kit/pull/3814

It's in an unfinished state, but I think it demonstrates the approach pretty well. I'm not convinced the added complexity and necessary changes to headless internals are worth it.

Reply:
I understand that allowing multiple listing / search interfaces to be supported in the same engine would require minimal effort, and I appreciate the elegance of the proposed solution.

However, what I'm hesitant about is that it would be making it easy to implement odd things that I believe should only be done exceptionally, if at all. This could confuse people into thinking that they _should_ typically be displaying multiple parallel listing / search interfaces, which is definitely not what we want.

If someone thinks that they need two product listing pages to be displayed simultaneously, I personally believe we should challenge that idea because, well, it's weird and unusual. If they still really want to do it but we still feel it's an edge case, we could always tell them to use two engines. If they do manage to convince us that this is a legitimate and common enough use case (or if a sufficient number of people ask for this feature), then we could consider allowing it for everyone. But I think our core assumption should be that this is weird and should normally not happen.

Now regarding facets and sort, I think we only need them to be at one place in the state. A recommendation interface should never have facets or sorting, and only one product listing or search interface should ever be displayed at any given time.

So, I think the issue we need to solve, really, is pagination, because there can and will be multiple paginated interfaces displayed simultaneously. One very simple and pragmatic way we could deal with this would be to not use the pagination controller on recommendation interfaces at all, and instead make pagination a core feature of the recommendation controller. This way we'd have a set of recommendations, each with its own pagination.

Reply:
Excellent point! I 100% agree. I'll have a go at your proposed approach!

Reply:
@louis-bompart having discussed this briefly live with you I know that you don't 100% agree with the idea of having pagination as a core feature of the recommendations controller, can you elaborate on your position?

Reply:
I agree with most of https://github.com/coveo/ui-kit/pull/3808#discussion_r1570857058, apart from the solution proposed in the last paragraph.
In my opinion, it's still important to keep things divided because if a user, for example, needs to play with the pagination of its recs; I do think it's best if they can do so with a dedicated controller instead of being able to control the whole recs. 

So for the state, I don't think it changes anything, we're still using `state.commerceRecommendations[solutionTypeId].commercePagination`
However, it's modified with a dedicated controller (and ofc each instance of the controller can only access/modify the pagination of one solution).

Reply:
☝️ Gave this approach a try in https://github.com/coveo/ui-kit/pull/3825